io_buffer.c

```diff
@@ -2528,6 +2531,31 @@ rb_io_buffer_pread(VALUE self, VALUE io, rb_off_t from, size_t length, size_t of
     return rb_thread_io_blocking_region(io_buffer_pread_internal, &argument, descriptor);
 }
 
+/*
+ *  call-seq: pread(io, from, length, [offset]) -> numbytes
+ *
+ *  Read at most +length+ bytes from +io+ into the buffer, starting at
+ *  +from+, and put it in buffer starting from specified +offset+.
+ *
+ *  If +offset+ is not given, put it at the beginning of the buffer.
```

Comment:
I would suggest adding that the output file will be zero filled if the offset is beyond the end of the current file size.